src/main/java/com/nha/abdm/wrapper/hip/hrp/share/ProfileShareService.java

```diff
@@ -0,0 +1,101 @@
+  private final PatientRepo patientRepo;
+  private final RequestManager requestManager;
+  private final HIPClient hipClient;
+  private final RequestLogService requestLogService;
+  private final PatientService patientService;
+  @Autowired TokenNumberGenerator tokenNumberGenerator;
+
+  @Value("${profileOnSharePath}")
+  public String profileOnSharePath;
+
+  private static final Logger log = LogManager.getLogger(ProfileShareService.class);
+
+  public ProfileShareService(
+      PatientRepo patientRepo,
+      RequestManager requestManager,
+      HIPClient hipClient,
+      RequestLogService requestLogService,
+      PatientService patientService) {
+    this.patientRepo = patientRepo;
+    this.requestManager = requestManager;
+    this.hipClient = hipClient;
+    this.requestLogService = requestLogService;
+    this.patientService = patientService;
+  }
+
+  @Override
+  public void shareProfile(ProfileShare profileShare) {
+    String token = tokenNumberGenerator.generateTokenNumber();
+    log.info("Making post request to HIP-profile/share with token : " + token);
+    ResponseEntity<ProfileAcknowledgement> profileAcknowledgement =
+        hipClient.shareProfile(
+            ShareProfileRequest.builder()
+                .token(token)
+                .hipId(profileShare.getProfile().getHipCode())
+                .profile(profileShare)
+                .build());
+    ProfileAcknowledgement acknowledgement = profileAcknowledgement.getBody();
+    if (acknowledgement != null && acknowledgement.getStatus().equals("SUCCESS")) {
+      ProfileOnShare profileOnShare =
+          ProfileOnShare.builder()
+              .requestId(UUID.randomUUID().toString())
+              .resp(new RespRequest(profileShare.getRequestId()))
+              .timestamp(Utils.getCurrentTimeStamp())
+              .acknowledgement(acknowledgement)
+              .build();
+      log.info("onShare : " + profileOnShare.toString());
+      if (patientRepo.findByAbhaAddress(profileShare.getProfile().getPatient().getHealthId())
+          == null) {
+        Patient patient = new Patient();
+        patient.setAbhaAddress(profileShare.getProfile().getPatient().getHealthId());
+        patient.setGender(profileShare.getProfile().getPatient().getGender());
+        patient.setName(profileShare.getProfile().getPatient().getName());
+        patient.setDateOfBirth(
+            profileShare.getProfile().getPatient().getYearOfBirth()
+                + "-"
+                + profileShare.getProfile().getPatient().getMonthOfBirth()
+                + "-"
+                + profileShare.getProfile().getPatient().getDayOfBirth());
+        patient.setPatientDisplay(profileShare.getProfile().getPatient().getName());
+        patient.setPatientMobile(
+            profileShare.getProfile().getPatient().getIdentifiers().get(0).getValue());
+        patient.setEntity(profileShare.getProfile().getHipCode());
+        patientRepo.save(patient);
```

Comment:
repo save is not recommended as it inserts duplicates. please use PatientService's upsertPatients method

Reply:
After creating the object, adding the patient into db via patientService.upsertPatients(Collections.singletonList(patient));